cmgr/types.go

```diff
@@ -60,20 +60,26 @@ type HostInfo struct {
 }
 
 type NetworkOptions struct {
-	Internal bool `json:"internal"`
+	InternalNetwork bool `json:"internal_network" yaml:"internal_network"`
 }
 
 type ContainerOptions struct {
-	Init            bool     `json:"init,omitempty"`
-	Cpus            string   `json:"cpus,omitempty"`
-	Memory          string   `json:"memory,omitempty"`
-	Ulimits         []string `json:"ulimits,omitempty"`
-	PidsLimit       int64    `json:"pidslimit,omitempty"`
-	ReadonlyRootfs  bool     `json:"readonlyrootfs,omitempty"`
-	DroppedCaps     []string `json:"droppedcaps,omitempty"`
-	NoNewPrivileges bool     `json:"nonewprivileges,omitempty"`
-	DiskQuota       string   `json:"diskquota,omitempty"`
-	CgroupParent    string   `json:"cgroupparent,omitempty"`
+	Init            bool     `json:"init,omitempty"            yaml:"init"`
+	Cpus            string   `json:"cpus,omitempty"            yaml:"cpus"`
+	Memory          string   `json:"memory,omitempty"          yaml:"memory"`
+	Ulimits         []string `json:"ulimits,omitempty"         yaml:"ulimits"`
+	PidsLimit       int64    `json:"pidslimit,omitempty"       yaml:"pidslimit"`
+	ReadonlyRootfs  bool     `json:"readonlyrootfs,omitempty"  yaml:"readonlyrootfs"`
+	DroppedCaps     []string `json:"droppedcaps,omitempty"     yaml:"droppedcaps"`
+	NoNewPrivileges bool     `json:"nonewprivileges,omitempty" yaml:"nonewprivileges"`
+	DiskQuota       string   `json:"diskquota,omitempty"       yaml:"diskquota"`
+	CgroupParent    string   `json:"cgroupparent,omitempty"    yaml:"cgroupparent"`
+}
+
+type ChallengeOptions struct {
+	NetworkOptions   `yaml:",inline"`
+	ContainerOptions `yaml:",inline"`
+	Overrides        ContainerOptionsWrapper `json:"overrides,omitempty" yaml:"overrides"`
```

Comment:
Do you think it's potentially confusing that Overrides acts more like a full replacement?
Given the Compose-like syntax as a user I would expect them to merge, like given the example:
```yaml
init: true
cpus: 0.5
pidslimit: 5
overrides:
    work:
        pidslimit: 10
    randomDnsName:
        cpus: 0.25
```
there's the somewhat unintuitive result
| challenge                             | host          | init | cpus | pidslimit |
|---------------------------------------|---------------|------|----------|----|
| cmgr/examples/aptitude-and-privileges |               |    1 | 0.5  |          5 |
| cmgr/examples/aptitude-and-privileges | work          |    0 |  0 |    10 |
| cmgr/examples/aptitude-and-privileges | randomDnsName |    0 | 0.25 |          0 |

I thought about adding code to do the merge but unfortunately zero values make this complicated, for example:

```yaml
init: true
overrides:
    work:
        init: false  # no way to detect this vs. unset
```

I guess the Go solution for this is to make every struct field a pointer so null vs. zero value can be disambiguated, but that seems like it'll add a lot of complexity to handle in the DB load/save methods and when setting the Docker HostOptions.